tests/unit-tests/dds/DCPS/RTPS/RtpsCoreTypeSupportImpl.cpp

```diff
@@ -50,11 +52,59 @@ void extract_Parameter_user_data(const std::string& user_data, const Encoding en
   }
 }
 
+// ---------- ---------- ---------- ---------- ----------
+template <typename Seq, typename Element>
+bool insert_sequence(Serializer& s, const std::vector<Element> v)
+{
+  const CORBA::ULong length = static_cast<CORBA::ULong>(v.size());
+  Seq seq;
+  seq.length(length);
+  for (CORBA::ULong i = 0; i < length; ++i) {
+    seq[i] = v[i];
+  }
+  return s << seq;
+}
+
+template <typename Seq, typename Element>
+void extract_sequence(Serializer& s, const std::vector<Element> v)
+{
+  const CORBA::ULong length = static_cast<CORBA::ULong>(v.size());
+  Seq seq;
+  ASSERT_TRUE(s >> seq);
+  ASSERT_EQ(seq.length(), length);
+  for (CORBA::ULong i = 0; i < length; ++i) {
+    EXPECT_EQ(seq[i], v[i]);
+  }
+}
+
+template <typename Seq>
+bool extract_sequence_with_unverified_length(const CORBA::ULong length = 0)
+{
+  const CORBA::ULong data[] = {length, 1000};
+  ACE_Message_Block mb(512);
+  mb.copy((const char*)data, sizeof(data));
+  Serializer s(&mb, Encoding(Encoding::KIND_XCDR1));
+
+  Seq seq;
+  return s >> seq;
+}
+
+// ---------- ---------- ---------- ---------- ----------
+void extract_FilterResult_t(const std::vector<CORBA::Long> v)
+{
+  const CORBA::ULong length = static_cast<CORBA::ULong>(v.size());
+  ACE_Message_Block mb(length * 32 + 128);
```

Comment:
Where does this expression come from?  Why is it 32 and 128?

Reply:
They are both used roughly to provide enough space in this test, instead of using the same sized block such as 1024. '32' is for each element (may be increased if bigger elements are used). '123' is an extra space for the sequence, including the zero length cases.

Reply:
32 bytes is 8 times the sizes of a 4 byte ULong. It could use `serialized_size` on the sequence to get the exact size it requires.

Reply:
You are both right. Since this is not a generic sequence extracting function, there is no need to provide too much extra space for each element.